Modules/BookingManager/Reservations/class.ilBookingReservationsGUI.php

```diff
@@ -506,7 +504,7 @@ public function rsvCancel() : void
                         $ids[] = $aggr_ids;
                     } else {
                         $aggr_ids = array_slice($aggr_ids, 0, $max);
-                        $ids = array_merge($ids, $aggr_ids);
+                        $ids = array_merge($ids, $aggr_ids);//PHP8Review: This may cause performance issues inside of a foreach
```

Comment:
@alex40724 Im not fully aware what you mean with that.
Im not concerned about the looping index of the foreach itself but about the dynmacly called array_merge, since array_merge has a very bad perfromance in general, sepacially for the purpose addition because i might reindex the whole $root array (not just the added part) in every call depending if its key a numeric.
Of course this heavily depends on the array composition and size, but since i couldnt find a proof of associative keys or its size i worte this hint in concern. Maybe you are more aware of the circumstances and could provide me with that info.

In both ways it would be an improved (or even) solution to stack the array and marge them after the loop, since you dont need them in it (since this ensure only one reindexing):
```
$stack = [];
for($map as $a) {
   //do something with $a
  $stack[] = $a;
}

array_merge(...$stack);
```